src/grafana_service.ts

```diff
@@ -1,16 +1,24 @@
 import { Metric } from './metrics/metrics_factory';
-import { MetricQuery } from './metrics/metric';
+import { MetricQuery, Datasource } from './metrics/metric';
 
 import { URL } from 'url';
 import axios from 'axios';
 import * as _ from 'lodash';
 
-
-export class GrafanaUnavailable extends Error {};
-export class DatasourceUnavailable extends Error {
-  public url: string;
+export class DataKitError extends Error {
+  constructor(
+    message: string,
+    public datasourceType?: string,
```

Comment:
do we have an `enum` for `datasourceType` ?

Reply:
oh, I see. Let's make it ones: https://github.com/CorpGlory/grafana-datasource-kit/blob/master/src/metrics/metric.ts#L3

Reply:
added issue #78